django/core/checks/translation.py

```diff
@@ -0,0 +1,24 @@
+import re
+
+from django.conf import settings
+
+from . import Tags, Warning, register
+
+LANGUAGE_CODE_PATTERN = re.compile(r'^[a-z]{2}(-[a-z]{2})?$')
+
+
+@register(Tags.translation)
+def check_setting_language_code(app_configs, **kwargs):
+    """
+    Warn if language code is in the wrong format. Language codes are generally represented in lower-case with a dash
+    separator.
+    """
+    match_result = re.match(LANGUAGE_CODE_PATTERN, settings.LANGUAGE_CODE)
+    errors = []
+    if not match_result:
+        errors.append(Warning(
```

Comment:
I think I'd make this an `Error` - I don't think translation works at all if this is wrong.

Reply:
Done 👍